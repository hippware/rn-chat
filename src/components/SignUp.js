// @flow

import React from 'react';
import {View, Image, StyleSheet, Text, Linking} from 'react-native';
import {Actions} from 'react-native-router-flux';
import {k} from './Global';
import {StatelessForm} from 'react-native-stateless-form';
import SignUpTextInput from './SignUpTextInput';
import SignUpAvatar from './SignUpAvatar';
import model from '../model/model';
import {observer} from 'mobx-react/native';
import * as log from '../utils/log';
import {observable} from 'mobx';
import autobind from 'autobind-decorator';
import {colors} from '../constants';
import Button from 'apsl-react-native-button';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
import profileStore from '../store/profileStore';

@autobind
@observer
export default class SignUp extends React.Component {
  componentDidMount() {
    // set correct isValid
    if (model.profile) {
      try {
        model.profile.validate();
      } catch (e) {
      }
    }
  }
  @observable loading: boolean = false;
  render() {
    if (!model.profile) {
      log.log('NULL PROFILE!', {level: log.levels.ERROR});
      return null;
    }
    const {loaded, handle, user} = model.profile;
    if (!loaded) {
      log.log('PROFILE IS NOT LOADED', handle, user, {level: log.levels.ERROR});
    }
    return (
      <KeyboardAwareScrollView keyboardShouldPersistTaps='always'>
        <StatelessForm>
          <View style={{marginLeft: 70 * k, marginRight: 70 * k, marginTop: 47.5 * k, flexDirection: 'row'}}>
            <Image style={{width: 60 * k, height: 69 * k}} source={require('../../images/pink.png')} />
            <View style={{paddingLeft: 20 * k}}>
              <Text style={{fontFamily: 'Roboto-Light', fontSize: 30 * k, color: colors.PINK, lineHeight: 35 * k}}>Let's create your profile!</Text>
            </View>
          </View>
          <View style={{marginTop: 15 * k, marginBottom: 15 * k, alignItems: 'center'}}>
            <SignUpAvatar avatar={model.profile.avatar} />
          </View>
          <SignUpTextInput icon={require('../../images/iconUsernameNew.png')} name='handle' data={model.profile} label='Username' autoCapitalize='none' />
          <SignUpTextInput icon={require('../../images/iconSubsNew.png')} name='firstName' data={model.profile} label='First Name' />
          <SignUpTextInput name='lastName' data={model.profile} label='Last Name' />
          <SignUpTextInput
            onSubmit={Actions.states.signUp.success}
            icon={require('../../images/iconEmailNew.png')}
            name='email'
            data={model.profile}
            label='Email'
            autoCapitalize='none'
          />
          <Text style={styles.agreeNote}>
            {'By signing up you agree to our '}
            <Text onPress={() => Linking.openURL('https://tinyrobot.com/privacy-policy/')} style={styles.linkText}>
              Privacy Policy
            </Text>
            {',\r\n '}
            <Text onPress={() => Linking.openURL('https://tinyrobot.com/terms-of-service/')} style={styles.linkText}>
              Terms of Service
            </Text>
            <Text>{', and for us to contact you via email\r\nfor updates and information.'}</Text>
          </Text>
          <Button
            isLoading={Actions.currentScene !== this.props.name}
            isDisabled={!model.profile.isValid}
            onPress={() => {
              profileStore.isNew = true;
              Actions.states.signUp.success();
            }}
            style={styles.submitButton}
            textStyle={styles.text}
          >
            Done
          </Button>
        </StatelessForm>
      </KeyboardAwareScrollView>
    );
  }
}

const styles = StyleSheet.create({
  text: {fontSize: 17.5 * k, letterSpacing: 0.8, fontFamily: 'Roboto-Regular', color: 'white'},
  submitButton: {
    marginLeft: 37.5 * k,
    marginRight: 37.5 * k,
    borderRadius: 4 * k,
    height: 50 * k,
    borderWidth: 0,
    backgroundColor: 'rgb(254,92,108)',
  },
  agreeNote: {
    marginTop: 35 * k,
    marginBottom: 35 * k,
    fontSize: 12.5 * k,
    color: colors.DARK_GREY,
    fontFamily: 'Roboto-Regular',
    textAlign: 'center',
  },
  linkText: {
    fontSize: 12.5 * k,
    color: colors.DARK_GREY,
    fontFamily: 'Roboto-Bold',
  },
  paginationStyle: {bottom: 170 * k},
});
