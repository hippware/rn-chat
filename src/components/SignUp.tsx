import React from 'react'
import {View, Image, StyleSheet, Text, Linking, TouchableOpacity} from 'react-native'
import {observable, runInAction} from 'mobx'
import {observer, inject} from 'mobx-react/native'
import {Actions} from 'react-native-router-flux'
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view'
import {k, avatarScale, minHeight, height} from './Global'
import FormTextInput from './FormTextInput'
import {colors} from '../constants'
import {RText, Spinner} from './common'
import {ValidatableProfile} from '../utils/formValidation'
import {IWocky} from 'wocky-client'
import {getSnapshot} from 'mobx-state-tree'
import alert from '../utils/alert'

type Props = {
  wocky?: IWocky
  analytics?: any
  warn?: any
}

@inject('wocky', 'analytics', 'warn')
@observer
class SignUp extends React.Component<Props> {
  @observable vProfile: ValidatableProfile | null = null
  handle: any
  firstName: any
  lastName: any
  email: any
  when: any

  componentDidMount() {
    runInAction(
      () =>
        (this.vProfile =
          this.props.wocky!.profile && new ValidatableProfile(this.props.wocky!.profile!))
    )
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.routeName === 'signUp') {
      runInAction(
        () =>
          (this.vProfile =
            this.props.wocky!.profile && new ValidatableProfile(this.props.wocky!.profile!))
      )
    }
  }

  done = async () => {
    if (!this.vProfile!.isValid) return
    const {profile} = this.props.wocky!
    if (!profile || profile!.updating) return
    try {
      await profile!.update(this.vProfile!.asObject)
      Actions.checkOnboarded()
      this.props.analytics.track('createprofile_complete', {
        profile: getSnapshot(this.props.wocky!.profile!),
      })
    } catch (err) {
      // notificationStore is probably not good here, user must read and confirm
      try {
        // display first error if it is GraphQL error
        const error = JSON.parse(err.message)[0].message
        alert(null, error)
      } catch {
        alert(null, err.message)
      }
      this.props.analytics.track('createprofile_fail', {
        profile: this.vProfile!.asObject, // send entered data to mixpanel, not original profile
        error: err,
      })
    }
  }

  render() {
    const {profile} = this.props.wocky!
    if (!profile) {
      return (
        <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
          <Text>Profile is not loaded</Text>
        </View>
      )
    }
    const buttonDisabled = (this.vProfile && !this.vProfile.isValid) || profile.updating
    return (
      <KeyboardAwareScrollView>
        <View
          style={{
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            height,
          }}
        >
          <View
            style={{
              marginTop: 50 * minHeight,
              flexDirection: 'row',
              justifyContent: 'center',
              alignItems: 'center',
            }}
            testID="signUpTopRow"
          >
            <Image
              style={{width: 60 * avatarScale, height: 69 * avatarScale}}
              source={require('../../images/pink.png')}
            />
            <View style={{paddingLeft: 20 * k}}>
              <RText weight="Light" size={30} color={colors.PINK} style={{height: 80}}>
                Let's get{'\r\n'}started!
              </RText>
            </View>
          </View>
          <View
            style={{
              justifyContent: 'center',
              alignSelf: 'center',
              width: '80%',
              marginVertical: 80 * minHeight,
            }}
          >
            <FormTextInput
              icon={require('../../images/iconUsernameNew.png')}
              ref={r => (this.handle = r)}
              label="Username"
              autoCapitalize="none"
              onSubmitEditing={() => this.firstName.focus()}
              store={this.vProfile ? this.vProfile.handle : undefined}
              testID="signUpUsername"
            />
          </View>
          <RText
            size={12.5}
            color={colors.DARK_GREY}
            style={[styles.agreeNote as any, {paddingHorizontal: 35}]}
          >
            {'By signing up you agree to our '}
            <RText
              size={12.5}
              weight="Bold"
              onPress={() => Linking.openURL('https://tinyrobot.com/privacy-policy/')}
            >
              Privacy Policy
            </RText>
            {', \r\n'}
            <RText
              size={12.5}
              weight="Bold"
              onPress={() => Linking.openURL('https://tinyrobot.com/terms-of-service/')}
            >
              Terms of Service
            </RText>
            <RText size={12.5} color={colors.DARK_GREY} style={styles.agreeNote as any}>
              {', and for us to contact you via email for updates and information.'}
            </RText>
          </RText>
          <TouchableOpacity
            disabled={buttonDisabled || !this.props.wocky!.connected}
            onPress={this.done}
            style={styles.submitButton}
          >
            <RText size={17.5} color="white" style={{letterSpacing: 0.8, textAlign: 'center'}}>
              {profile.updating ? <Spinner color="white" size={22} /> : 'Done'}
            </RText>
          </TouchableOpacity>
        </View>
      </KeyboardAwareScrollView>
    )
  }
}

export default SignUp

const styles = StyleSheet.create({
  submitButton: {
    borderRadius: 6,
    height: 50 * k,
    backgroundColor: 'rgb(254,92,108)',
    width: '80%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  agreeNote: {
    marginTop: 35 * k,
    marginBottom: 35 * k,
    textAlign: 'center',
  },
  paginationStyle: {bottom: 170 * k},
})
