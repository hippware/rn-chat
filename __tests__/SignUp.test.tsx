import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SignUp from '../src/components/SignUp'
import {Provider} from 'mobx-react/native'

describe('Signup', () => {
  it('renders', () => {
    const wocky = {
      createChat: () => {
        /*noop*/
      },
      profile: {
        handle: 'jerkham',
        firstName: 'eric',
        lastName: 'kirkham',
        email: 'eric.kirkham@gmail.com',
        loaded: true,
        updateError: '',
      },
    }
    const analytics = {
      track: () => {
        /*noop*/
      },
    }
    const profileValidationStore = {
      setProfile: () => {
        /*noop*/
      },
    }
    const warn = () => {
      /*noop*/
    }
    const tree = renderer
      .create(
        <Provider
          wocky={wocky}
          profileValidationStore={profileValidationStore}
          analytics={analytics}
          warn={warn}
        >
          <SignUp />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
