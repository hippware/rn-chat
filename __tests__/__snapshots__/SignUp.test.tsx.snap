// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginLeft": 140,
          "marginRight": 140,
          "marginTop": 95,
        }
      }
      testID="signUpTopRow"
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/pink.png",
          }
        }
        style={
          Object {
            "height": 138,
            "width": 120,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(253,95,108)",
                "fontFamily": "Roboto-Light",
                "fontSize": 30,
              },
              Object {
                "lineHeight": 70,
              },
            ]
          }
        >
          Let's create your profile!
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 30,
          "marginTop": 30,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View>
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../images/addPhoto.png",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 72,
                  "height": 144,
                  "width": 144,
                },
                undefined,
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 72,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingBottom": 13,
                "paddingHorizontal": 10,
                "paddingTop": 9,
              },
              Object {
                "justifyContent": "center",
                "paddingTop": 2,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/iconUsernameNew.png",
                }
              }
              style={
                Array [
                  Object {
                    "marginLeft": 16,
                    "marginRight": 13,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              clearButtonMode="while-editing"
              icon={
                Object {
                  "testUri": "../../../images/iconUsernameNew.png",
                }
              }
              label="Username"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Username"
              placeholderTextColor="rgb(212,212,212)"
              returnKeyType="next"
              store={
                ValidateItem {
                  "errorMessage": "",
                  "isValid": undefined,
                  "key": "handle",
                  "value": "southerneer",
                  Symbol(mobx did run lazy initializers): true,
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "ValidateItem@13.keys",
                      "observers": Set {},
                    },
                    "name": "ValidateItem@13",
                    "target": [Circular],
                    "values": Map {
                      "errorMessage" => "",
                      "value" => "southerneer",
                      "isValid" => undefined,
                    },
                  },
                }
              }
              style={
                Object {
                  "color": "rgb(63,50,77)",
                  "flex": 1,
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 18,
                }
              }
              testID="signUpUsername"
              underlineColorAndroid="transparent"
              value="southerneer"
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": 30,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(63, 50, 77, .2)",
                "height": 0.5,
              },
              undefined,
            ]
          }
        />
      </View>
      <View>
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingBottom": 13,
                "paddingHorizontal": 10,
                "paddingTop": 9,
              },
              Object {
                "justifyContent": "center",
                "paddingTop": 2,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/iconSubsNew.png",
                }
              }
              style={
                Array [
                  Object {
                    "marginLeft": 16,
                    "marginRight": 13,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              clearButtonMode="while-editing"
              icon={
                Object {
                  "testUri": "../../../images/iconSubsNew.png",
                }
              }
              label="First Name"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="First Name"
              placeholderTextColor="rgb(212,212,212)"
              returnKeyType="next"
              store={
                ValidateItem {
                  "errorMessage": "",
                  "isValid": undefined,
                  "key": "firstName",
                  "value": "eric",
                  Symbol(mobx did run lazy initializers): true,
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "ValidateItem@18.keys",
                      "observers": Set {},
                    },
                    "name": "ValidateItem@18",
                    "target": [Circular],
                    "values": Map {
                      "errorMessage" => "",
                      "value" => "eric",
                      "isValid" => undefined,
                    },
                  },
                }
              }
              style={
                Object {
                  "color": "rgb(63,50,77)",
                  "flex": 1,
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 18,
                }
              }
              testID="signUpFirstName"
              underlineColorAndroid="transparent"
              value="eric"
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": 30,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(63, 50, 77, .2)",
                "height": 0.5,
              },
              undefined,
            ]
          }
        />
      </View>
      <View>
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingBottom": 13,
                "paddingHorizontal": 10,
                "paddingTop": 9,
              },
              Object {
                "justifyContent": "center",
                "paddingTop": 2,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "width": 40,
                }
              }
            />
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              clearButtonMode="while-editing"
              label="Last Name"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Last Name"
              placeholderTextColor="rgb(212,212,212)"
              returnKeyType="next"
              store={
                ValidateItem {
                  "errorMessage": "",
                  "isValid": undefined,
                  "key": "lastName",
                  "value": "kirkham",
                  Symbol(mobx did run lazy initializers): true,
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "ValidateItem@21.keys",
                      "observers": Set {},
                    },
                    "name": "ValidateItem@21",
                    "target": [Circular],
                    "values": Map {
                      "errorMessage" => "",
                      "value" => "kirkham",
                      "isValid" => undefined,
                    },
                  },
                }
              }
              style={
                Object {
                  "color": "rgb(63,50,77)",
                  "flex": 1,
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 18,
                }
              }
              testID="signUpLastName"
              underlineColorAndroid="transparent"
              value="kirkham"
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": 30,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(63, 50, 77, .2)",
                "height": 0.5,
              },
              undefined,
            ]
          }
        />
      </View>
      <View>
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingBottom": 13,
                "paddingHorizontal": 10,
                "paddingTop": 9,
              },
              Object {
                "justifyContent": "center",
                "paddingTop": 2,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/iconEmailNew.png",
                }
              }
              style={
                Array [
                  Object {
                    "marginLeft": 16,
                    "marginRight": 13,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              clearButtonMode="while-editing"
              icon={
                Object {
                  "testUri": "../../../images/iconEmailNew.png",
                }
              }
              keyboardType="email-address"
              label="Email"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Email"
              placeholderTextColor="rgb(212,212,212)"
              returnKeyType="done"
              store={
                ValidateItem {
                  "errorMessage": "",
                  "isValid": undefined,
                  "key": "email",
                  "value": "eric@hippware.com",
                  Symbol(mobx did run lazy initializers): true,
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "ValidateItem@24.keys",
                      "observers": Set {},
                    },
                    "name": "ValidateItem@24",
                    "target": [Circular],
                    "values": Map {
                      "errorMessage" => "",
                      "value" => "eric@hippware.com",
                      "isValid" => undefined,
                    },
                  },
                }
              }
              style={
                Object {
                  "color": "rgb(63,50,77)",
                  "flex": 1,
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 18,
                }
              }
              testID="signUpEmail"
              underlineColorAndroid="transparent"
              value="eric@hippware.com"
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": 30,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(63, 50, 77, .2)",
                "height": 0.5,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(155,155,155)",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12.5,
          },
          Object {
            "fontSize": 25,
            "marginBottom": 70,
            "marginTop": 70,
            "textAlign": "center",
          },
        ]
      }
    >
      By signing up you agree to our 
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "Roboto-Bold",
              "fontSize": 12,
            },
            undefined,
          ]
        }
      >
        Privacy Policy
      </Text>
      ,
 
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "Roboto-Bold",
              "fontSize": 12,
            },
            undefined,
          ]
        }
      >
        Terms of Service
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "Roboto-Regular",
              "fontSize": 12,
            },
            undefined,
          ]
        }
      >
        , and for us to contact you via email
for updates and information.
      </Text>
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
          },
          Object {
            "backgroundColor": "rgb(254,92,108)",
            "borderRadius": 8,
            "borderWidth": 0,
            "height": 100,
            "marginLeft": 75,
            "marginRight": 75,
          },
          Object {
            "opacity": 0.5,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
              "fontSize": 18,
              "textAlign": "center",
            },
            Object {
              "color": "white",
              "fontFamily": "Roboto-Regular",
              "fontSize": 35,
              "letterSpacing": 0.8,
            },
          ]
        }
      >
        Done
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
